pkg/roachpb/tenant.go

```diff
@@ -223,3 +224,83 @@ func (c *TenantNameContainer) Get() TenantName {
 func (c *TenantNameContainer) String() string {
 	return (*syncutil.AtomicString)(c).Get()
 }
+
+// TenantIdentity is an interface that represents a tenant's identity. Both
+// TenantID and TenantName implement this interface.
+type TenantIdentity interface {
+	// IsSet returns whether this tenant identity is set or not.
+	IsSet() bool
+	// IsValid returns an error if the tenant identity is invalid.
+	IsValid() error
+	// IsSystem returns whether this tenant identity is that of the system tenant.
+	IsSystem() bool
+	// IsEqual returns whether this tenant identity is equal to another.
+	IsEqual(o TenantIdentity) bool
+	// ToString returns a string representation of the tenant identity.
+	ToString() string
```

Comment:
If you make this `String()` rather than `ToString()` then this interface is also a fmt.Stringer which can be useful https://pkg.go.dev/fmt#Stringer